plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -309,6 +309,13 @@ private function init_hooks() {
 		add_action( 'woocommerce_updated', array( $this, 'add_woocommerce_remote_variant' ) );
 		add_action( 'woocommerce_newly_installed', 'wc_set_hooked_blocks_version', 10 );
 		add_action( 'update_option_woocommerce_allow_tracking', array( $this, 'get_tracking_history' ), 10, 2 );
+		// phpcs:ignore Generic.Commenting.Todo.TaskFound
+		// @todo: remove this once we have version 3.9.3 of Action Scheduler.
+		if ( function_exists( 'as_supports' ) && as_supports( 'schedule_recurring_actions_hook' ) ) {
+			add_action( 'action_scheduler_schedule_recurring_actions', array( $this, 'register_recurring_actions' ) );
+		} else {
+			add_action( 'action_scheduler_init', array( $this, 'register_recurring_actions' ) );
```

Comment:
This will make the queries attempting to schedule the unique actions run on every request which can cause quite a bit of overhead.

Our options are probably to:
1. Wait until the latest version of AS is included and schedule_recurring_actions_hook is supported
2. Only run this on install or updates - recurring actions should stay scheduled unless one of them fail for some unrelated reason, which carries some risk.
3. Do step 2 and also run it on one of the built in crons, e.g. twicedaily.